compiler/noirc_frontend/src/signed_field.rs

```diff
@@ -183,15 +199,33 @@ impl PartialOrd for SignedField {
     }
 }
 
-impl From<FieldElement> for SignedField {
-    fn from(value: FieldElement) -> Self {
-        Self::new(value, false)
+impl From<u32> for SignedField {
+    fn from(value: u32) -> Self {
+        Self::new(value.into(), false)
     }
 }
 
-impl From<SignedField> for FieldElement {
-    fn from(value: SignedField) -> Self {
-        if value.is_negative { -value.field } else { value.field }
+impl From<u128> for SignedField {
+    fn from(value: u128) -> Self {
+        Self::new(value.into(), false)
+    }
+}
+
+impl From<i128> for SignedField {
+    fn from(value: i128) -> Self {
+        if value < 0 { Self::new((-value).into(), true) } else { Self::new(value.into(), false) }
```

Comment:
This will panic when `value = -128`

Reply:
Good catch! (I guess you meant `i128::MIN`, not sure! I added a couple of tests)

Reply:
Oops, yes I saw `i128` and thought `i8::MIN = -128` for some reason, but yes the minimum value.

Reply:
Was this fine? I'll add a few tests for it tomorrow just in case.